File: 0190-reverse-bits/0190-reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        b = bin(n)[2:]
        # pad 0 on the left to make it 31 digit
        zeros = 31 - len(b)
        for _ in range(zeros):
            b = '0' + b
        # pad the sign bit
        # 0 for positive
        # 1 for negative
        b = ('0' + b) if (n >= 0) else ('1' + b)

        m = 1 # to hold multiples of 2
        r = 0 # to hold the result
        for d in b:
            r += int(d) * m
            m *= 2
        return r
        

File: 0190-reverse-bits/0190-reverse-bits.js
/**
 * @param {number} n
 * @return {number}
 */
var reverseBits = function (n) {
    let b = n.toString(2)
    let l = 32 - b.length
    for(let i=1; i < l; ++i) {
        b = '0' + b;
    }
    b = (n >= 0) ? '0' + b : '1' + b 
    // console.log(b)
    let m = 1   // to hold 2^p
    let r = 0  // to hold the result
    for (let i = 0; i < b.length; ++i) {
        r += b[i] * m
        m *= 2
    }
    return r
};